pinot-broker/src/main/java/org/apache/pinot/broker/routing/adaptiveserverselector/AdaptiveServerSelector.java

```diff
@@ -32,27 +32,29 @@ public interface AdaptiveServerSelector {
    * Picks the best server to route a query from the list of candidate servers.
    *
    * @param serverCandidates Candidate servers from which the best server should be chosen.
-   * @return server identifier
+   * @return server identifier and whether server is online
    */
-  String select(List<String> serverCandidates);
+  Pair<String, Boolean> select(List<Pair<String, Boolean>> serverCandidates);
```

Comment:
The ONLINE/OFFLINE is for a single segment server state. Here we should probably only keep the online servers

Reply:
done